lambda/postSchedulePeerTeachers/Scheduler/src/main/java/db/Person.java

```diff
@@ -160,13 +161,34 @@ public boolean isGoodSection(int sectionId, Section section) {
         return isQualified(section.getCourseId()) && isAvailable(section.getMeetings()) && !getPreferenceEquals(sectionId, DBPreference.CANT_DO);
     }
 
+    public boolean alreadyAssigned(String weekday, Time start, Time end) {
+        for (Section section : sections.values()) {
+            for (Meeting meeting : section.getMeetings()) {
+                if (weekday.equals(meeting.getWeekday()) && !(meeting.getStartTime().after(end) || meeting.getEndTime().before(start))) {
```

Comment:
I don't know Java enough to know if this can be sped up. If this function is called a bunch though, I'd look into it

Reply:
https://stackoverflow.com/questions/47658552/how-to-find-any-element-in-a-hashmap-where-a-given-condition-about-that-element ??

Reply:
The link that you sent cannot optimize the function further. There is no way (with the current design) to speedup the function, but it is actually not as slow as it would seem.